src/distributions/uniform.rs

```diff
@@ -769,9 +779,12 @@ impl UniformSampler for UniformChar {
 
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Self::X {
         let mut x = self.sampler.sample(rng);
-        if x >= 0xD800 {
-            x += 0xE000 - 0xD800;
+        if x >= CHAR_SURROGATE_START {
+            x += CHAR_SURROGATE_LEN;
         }
+        // SAFETY: x must not be in surrogate range or greater than char::MAX.
+        // This relies on range constructors which accept char arguments.
+        // Validity of input char values is assumed.
         unsafe { core::char::from_u32_unchecked(x) }
```

Comment:
The safety relies on the implementation of a safe trait, so this can technically be UB. However, the corresponding implementation is in the same module, so I guess it's fine. Could you however please add a `debug_assert`?

Reply:
No point: if you look at `from_u32_unchecked` it has that assert already. The only way this can fail (other than code bugs here) is if an invalid `char` is used as input (meaning something else already violated this contract).